### ISSUE TITLE
Status is not updated when manual charging

### ISSUE THREAD

**Issue by abstractionnl** [jonasbkarlsson/ev_smart_charging#327]
There is currently no way of knowing manual charging is active or not. When pressing the manually start charging button, the charger is enabled and EV starts charging. However, the status of EV Smart charging either says "Not active" (if smart charging is disabled) or "Waiting for charging to start" (if planned).

It would be nice if the status switched to "Manual charging" to indicate manual charging override is currently active.

**Comment by Hennyr** [comment]
Good idea. But please add also when pricelimit, pricemax or oppotunistisch charge is activated and have status Wait/charging/complete…

**Comment by ronlimon** [comment]
We'd like to help with this, and it ties neatly into jonasbkarlsson/ev_smart_charging#450 / jonasbkarlsson/ev_smart_charging#458.

The generic primitive both need is one user-configured "charging state entity" — the user points EVSC at whatever their car or charger exposes as "charging now" (a binary_sensor, a status sensor, an enum). EVSC normalizes it to on/off. That single input powers both:

- **verify a commanded start actually took effect** (jonasbkarlsson/ev_smart_charging#450 / jonasbkarlsson/ev_smart_charging#458), and
- **detect manual/external charging** — actual-charging ON while EVSC didn't command it → a "Manual charging" status (this issue). That also covers what PR jonasbkarlsson/ev_smart_charging#447 doesn't yet: charging started entirely outside EVSC (e.g. an RFID tap at the charger), not just via EVSC's own manual button.

We'd avoid hardcoding per-car logic — every integration reports charging differently and renames things over time. Instead: user-selected entity + small normalization for common state strings + a community-maintained "common cars" mapping table in the README, which we'd seed with our car (Zeekr 7X) as the first worked example. Car-agnostic and low-maintenance.

Happy to build this — unify it with jonasbkarlsson/ev_smart_charging#458 around one charging-state entity, with tests. Would that direction work for you, @jonasbkarlsson?

**Comment by jonasbkarlsson** [comment]
Frankly, one of the reasons why I have been so slow at adding new PRs lately is that I have started to feel that the integration is becoming too complex to handle. The manual charging function was intentionally implemented to be independently from the automatic charging.

But now I have merged the **verify a commanded start actually took effect** PR, so how should manual/external charging interact with that?

For example, the following sequence of events:
- Automatic charging starts
- The actual charging start fails
- A second charging start is done
- The second charging start is successful
- A manual (or external) stop charging is done
- The charging stops

Now, I think that, the **verify a commanded start actually took effect** function will start the charging again...

So the current implementation of **verify a commanded start actually took effect** is not good.

And in addition to the above example, there are several other sequences of events that one need to handle. Just to illustrate how complicated things are. So can someone state the expected interaction between the verify commanded start, manual/external charging and the automatic charging?

**Comment by ronlimon** [comment]
@jonasbkarlsson — first, kudos for naming the complexity, and for pushing back on your own merge. Recognising that verify-start × manual/external × automatic is a genuinely hard little state-machine — and that keeping manual charging deliberately independent was the right instinct — is worth more than any single feature. Your sequence is real:

> auto start → start fails → retried → success → manual/external stop → verify restarts it

The root ambiguity: `auto_state == ON && actual == OFF` can't tell *"commanded start silently failed"* (retry) apart from *"the user just stopped it"* (stand down). Resolving that means tracking **whether actual charging was ever confirmed since the last commanded start**, plus defining what counts as an external stop — and that state is exactly the kind of thing that makes an integration heavy.

Here's where we've landed, and why we think it argues **against** solving this inside EVSC:

We drive a wallbox that we command over OCPP (RemoteStart/Stop), with a car that can also pause/resume on its own. EVSC does what it's great at — decides *when* to charge from price look-ahead + SOC + target, and flips one boolean. A thin bridge automation on our side turns that decision into actual charge control, and it's that bridge — not EVSC — that owns the verify + manual question:

- a **"was this session HA-initiated?"** marker, set when the bridge commands a start, so a session someone begins by tapping their own RFID card at the charger is recognised as *external*, not a failed HA start;
- an **outcome check** a short delay after a commanded start, anchored on the charger's OCPP **connector status** + a real **power-draw** signal (our P1 meter) — not on a periodic "should be on but isn't → restart" poll;
- it separates *still transitioning* (PREPARING / FINISHING / SUSPENDED) from *really failed*, and the response is bounded (notify + at most one retry).

None of that belongs in EVSC: it leans on charger- and meter-side signals that vary enormously per setup, and it encodes site-specific policy (what an external stop *means* here). Pushing it upstream would make EVSC model charger-session and external-stop semantics for everyone — precisely the complexity you're wary of.

So our honest advice: keep EVSC's job "decide when to charge and command it," leave manual charging independent as you designed, and let each user reconcile verify/manual at their own charger/automation layer, where they actually have the right signals. If anything, one documented note — that verify-start assumes the commanded charger is the *only* thing controlling the plug — would save people the surprise.

On that basis I'll step back from the unify-around-one-charging-state-entity direction I floated earlier: for our setup EVSC already works well as-is, and we were really solving something that's better solved one layer down. Thanks for building and maintaining this. 🙏
